nilearn/decoding/tests/test_searchlight.py

```diff
@@ -175,3 +173,115 @@ def test_searchlight_group_cross_validation_with_extra_group_variable(
     # run searchlight on list of 3D images
     sl = searchlight.SearchLight(mask_img)
     sl.fit(imgs, y)
+
+
+def test_searchlight_attributes_exist_after_fit():
+    """Test if attributes `process_mask_` and `masked_scores_`
+    exist after fitting using mock data.
+    """
+    # Use the existing helper function to generate data
+    frames = 20
+    data_img, cond, mask_img = _make_searchlight_test_data(frames)
+
+    # Instantiate and fit the SearchLight with mock data
+    sl = searchlight.SearchLight(mask_img, radius=1.0)
+    sl.fit(data_img, y=cond)
+
+    # Check if attributes exist after fitting
+    assert hasattr(sl, "process_mask_"), "process_mask_ attribute missing."
+    assert hasattr(sl, "masked_scores_"), "masked_scores_ attribute missing."
+
+
+def test_searchlight_scores_img_error_before_fit():
+    """Test if accessing `scores_img_` raises an error before fitting."""
+    # Create mock mask
+    frames = 20
+    data_img, cond, mask_img = _make_searchlight_test_data(frames)
+
+    # Instantiate SearchLight without fitting
+    sl = searchlight.SearchLight(mask_img, radius=5.0)
+
+    # Check if accessing `scores_img_` raises a ValueError
+    with pytest.raises(ValueError, match="The model has not been fitted yet."):
+        sl.scores_img_
+
+
+def test_dimension_mismatch_error():
+    """Test if ValueError is raised when mask and image dimensions mismatch."""
+    data_img, cond, _ = _make_searchlight_test_data(frames=20)
+
+    # Create a mask with mismatched dimensions but valid content
+    invalid_mask_img = Nifti1Image(
+        np.ones((4, 4, 4), dtype="uint8"), np.eye(4)
+    )
+
+    sl = searchlight.SearchLight(invalid_mask_img, radius=1.0)
+
+    with pytest.raises(
+        ValueError, match="The mask image and the 4D input images must"
+    ):
+        sl.fit(data_img, y=cond)
+
+
+def test_access_scores_img_before_fit():
+    """Test if accessing `scores_img_` raises a ValueError before fitting."""
+    frames = 20
+    data_img, cond, mask_img = _make_searchlight_test_data(frames)
+    sl = searchlight.SearchLight(mask_img, radius=1.0)
+
+    with pytest.raises(ValueError, match="The model has not been fitted yet"):
+        _ = sl.scores_img_
+
+
+def test_transform_without_fit():
+    """Test if calling `transform()` raises ValueError before fitting."""
+    frames = 20
+    data_img, cond, mask_img = _make_searchlight_test_data(frames)
+    sl = searchlight.SearchLight(mask_img, radius=1.0)
+
+    with pytest.raises(ValueError, match="fit the model before calling"):
+        sl.transform(data_img)
+
+
+def test_transform_applies_mask_correctly():
+    """Test if `transform()` applies the mask correctly."""
+    frames = 20
+    data_img, cond, mask_img = _make_searchlight_test_data(frames)
+
+    sl = searchlight.SearchLight(mask_img, radius=1.0)
+    sl.fit(data_img, y=cond)
+
+    # Ensure model is fitted correctly
+    assert sl.scores_ is not None, "Scores were not set correctly during fit."
```

Comment:
```suggestion
    assert sl.scores_ is not None
```

let's use bare assert